Library/Homebrew/cmd/gist-logs.rb

```diff
@@ -84,13 +79,12 @@ def noecho_gets
     result
   end
 
-  def login(request)
+  def login!
     print "GitHub User: "
-    user = $stdin.gets.chomp
+    ENV["HOMEBREW_GITHUB_API_USERNAME"] = $stdin.gets.chomp
     print "Password: "
-    password = noecho_gets.chomp
+    ENV["HOMEBREW_GITHUB_API_PASSWORD"] = noecho_gets.chomp
     puts
-    request.basic_auth(user, password)
   end
```

Comment:
Any chance to drop these environment by put `login!` into Github module, so users won't set this env by their own.